### PR TITLE
Enable background sweep

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: Enable background sweep.

**Implementation Description (bullets)**: -

**Concerns (what feedback would you like?)**: Is there a less invasive way of enabling it?

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: -

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2104)

### CODE REVIEW

**Comment by fsamuel-bs** [comment]
I'll update docs and release notes shortly. The default parameters were tested in a manual run of sweep on a large internal customer, and should be sane for all use cases.

**Comment by gsheasby** [comment]
Fixes #1118. (I've been waiting for that for _months_)

This is a good way of doing it. Obviously needs messaging as you say, by docs (be sure to remove any super scary warnings), release notes and internal comms.

File: docs/source/cluster_management/sweep.rst [hunk 1]
```diff
@@ -45,34 +46,38 @@ Tunable Configuration Options
 -----------------------------
 
 The following optional parameters can be tuned to optimize Sweep performance for a specific AtlasDB instance.
-You may set them as part of your :ref:`AtlasDB configuration <atlas-config>`, or as a CLI option if you are running sweep using the CLI.
+You may set them as part of your :ref:`AtlasDB runtime configuration <atlas-config>`, or as a CLI option if you are running sweep using the CLI.
+Note that some of these parameters are just used as a hint. Sweep dynamically modifies these parameters according to successful or failed runs.
```
Comment (tboam) [inline docs/source/cluster_management/sweep.rst]: Are they a hint or are they the maximum value?  This implies that sweep may choose to exceed these values.

File: docs/source/cluster_management/sweep.rst [hunk 2]
```diff
@@ -45,34 +46,38 @@ Tunable Configuration Options
...
+   ``enabled``, "Only specified in config", "1,000", "Target number of (cell, timestamp) pairs to examine in a single run."
+   ``readLimit``, ``maxCellTsPairsToExamine``, "1,000", "Target number of (cell, timestamp) pairs to examine in a single run."
+   ``candidateBatchHint``, ``candidate-batch-hint``, "1", "Target number of candidate (cell, timestamp) pairs to load at once. Decrease this if sweep fails to complete (for example if the sweep job or the underlying KVS runs out of memory). Increasing it may improve sweep performance."
+   ``deleteBatchHint``, ``delete-batch-hint``, "1,000", "Target number of (cell, timestamp) pairs to delete in a single batch. Decrease if sweep cannot progress pass a large row or a large cell. Increasing it may improve sweep performance."
```
Comment (tboam) [inline docs/source/cluster_management/sweep.rst]: Do all these parameters appear in https://github.com/palantir/atlasdb/blob/develop/docs/source/cluster_management/sweep/sweep-endpoints.rst ?

Also looks like there's disagreement on the casing used by Endpoints between this doc and the sweep-endpoints.rst doc.

Comment (fsamuel-bs) [inline docs/source/cluster_management/sweep.rst]: Name and case of the parameters fixed.

File: docs/source/cluster_management/sweep.rst (hunk 1 above)
Comment (fsamuel-bs) [inline docs/source/cluster_management/sweep.rst]: It depends. There's some implementation details around each of them, so it's better if we announce them as a hint for now.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweeperServiceImpl.java [hunk 3]
```diff
@@ -76,10 +73,13 @@ public void sweepTableFromStartRowWithBatchConfig(String tableName,
...
-                        maxCellTsPairsToExamine == null ? DEFAULT_SWEEP_READ_LIMIT : maxCellTsPairsToExamine)
+                        maxCellTsPairsToExamine == null ? AtlasDbConstants.DEFAULT_SWEEP_READ_LIMIT :
+                                maxCellTsPairsToExamine)
                 .candidateBatchSize(
-                        candidateBatchSize == null ? DEFAULT_SWEEP_CANDIDATE_BATCH_HINT : candidateBatchSize)
-                .deleteBatchSize(deleteBatchSize == null ? DEFAULT_SWEEP_DELETE_BATCH_HINT : deleteBatchSize)
+                        candidateBatchSize == null ? AtlasDbConstants.DEFAULT_SWEEP_CANDIDATE_BATCH_HINT :
+                                candidateBatchSize)
+                .deleteBatchSize(deleteBatchSize == null ? AtlasDbConstants.DEFAULT_SWEEP_DELETE_BATCH_HINT :
+                        deleteBatchSize)
```
Comment (ashrayjain) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweeperServiceImpl.java]: The splitting here is weird. We usually follow something like
```
deleteBatchSize == null 
    ? AtlasDbConstants.DEFAULT_SWEEP_DELETE_BATCH_HINT
    : deleteBatchSize
```

**Pull request merged**